dscreator/sources/ferrybox/queries.py

```diff
@@ -47,43 +47,40 @@ def get_track(
     )
 
 
-@dataclass
-class TimeseriesResult:
-    uuid: str
-    values: List[str | int | float]
-    datetime: List[datetime]
-
-
 def get_ts(
     engine: Engine,
-    uuid: str,
+    track_uuid: str,
+    uuids: List[str],
     start_time: datetime,
     end_time: datetime,
-) -> TimeseriesResult:
+) -> Sequence[RowMapping]:
     """Query track
     The timeseries is limited to start_time<t<=end_time.
     """
     query = text(
         """
     SELECT
-        time, value
-    FROM
-        ts
+        track.time, 
+        ST_X(track.pos) as longitude, 
+        ST_Y(track.pos) as latitude,
+        ts.uuid, 
+        ts.value
+    FROM track
+    INNER JOIN ts 
+    ON track.time = ts.time
```

Comment:
This will not work in case of track time not being in full sync with the otehr boxes clock. 
I think there are data like that in NB and it was the reason why I have originally implemented it the way I did.